ccc/parameters.py

```diff
@@ -187,16 +187,14 @@ def update_specification(self, revision, raise_errors=True):
         if not (isinstance(revision, dict) or
                 isinstance(revision, str)):
             raise ValueError(
-                'ERROR: revision is not a dictionary of string')
-        if not revision:
-            return  # no revision to implement
+                'ERROR: revision is not a dictionary or string')
         self.adjust(revision, raise_errors=raise_errors)
         if self.errors and raise_errors:
```

Comment:
@hdoupe Can you help me out here.  It seems that if I have an error in the revision dictionary, it'll be caught in the `adjust` method from `ParamTools`.  Do I need this if statement?  I'm failing to write a test that catches it.

Reply:
Right, ParamTools will throw an error if the `revision` is not a dictionary. You should be fine without this if-statement since it's already covered in ParamTools. 

I need to double-check how user-friendly the error message is though.

Reply:
Great -  thanks @hdoupe !